packages/template-retail-react-app/app/hooks/use-active-data.js

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2023, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+/*global dw*/
+import {ACTIVE_DATA_ENABLE} from '@salesforce/retail-react-app/app/constants'
+
+const useActiveData = () => {
+    return {
+        async sendViewProduct(category, product, type) {
+            if (!ACTIVE_DATA_ENABLE) {
+                return
+            }
+            try {
+                if (dw?.ac) {
+                    if (category && category.id) {
+                        console.log('Applying category context')
+                        dw.ac.applyContext({category: category.id})
+                    }
+                    if (product && product.id) {
+                        console.log('Capturing product')
+                        dw.ac._capture({id: product.id, type: type})
+                    }
+                    if (dw.ac?._scheduleDataSubmission) {
+                        dw.ac._scheduleDataSubmission()
+                    }
+                }
+            } catch (err) {
+                console.error(err)
```

Comment:
```suggestion
                console.warn('Active Data request failed')
```
When an Einstein request fails, all that happens is "Einstein request failed" is printed as a warning. I think that these catch blocks should follow a similar pattern.